### PR TITLE
Python: parse all class methods

### PR DESCRIPTION
**Pull request by jdorweiler**
Two new things:

1. general overview for each module
https://beta.duckduckgo.com/?q=python+asyncio&t=ha&ia=about
selection_423

2. parse all methods for each class
https://beta.duckduckgo.com/?q=python+unittest+testloader.getTestCaseNames&t=ha&ia=about
selection_422

I didn't try to create a bunch of redirects. We can handle that in a separate PR. 

------
IA Page:  http://duck.co/ia/view/python 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @gaulrobe

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/python/parse.py [hunk 1]
```diff
@@ -141,6 +158,30 @@ def parse_for_method_signature(self, section):
...
+
+        Returns:
+            The method signature
+
+        """
+        id_tag = section.find('dt').get('id')
+        if id_tag:
+            tag_parts = id_tag.split('.')
+
+            # if it doesnt fit the patter
```
Comment (gaulrobe) [inline lib/fathead/python/parse.py]: Small typo here `pattern`

File: lib/fathead/python/parse.py [hunk 2]
```diff
@@ -141,6 +158,30 @@ def parse_for_method_signature(self, section):
...
+            #  module.class.method
+            # then concat the remaining parts into the method name
+            # ex: email.message.EmailMessage.is_attachment
+            if len(tag_parts) == 3:
+                return tag_parts
+            elif len(tag_parts) > 3:
+                return tag_parts[0], tag_parts[1], '.'.join(tag_parts[3:])
```
Comment (gaulrobe) [inline lib/fathead/python/parse.py]: Was [2] intentionally left out?

File: lib/fathead/python/parse.py [hunk 3]
```diff
@@ -159,23 +200,49 @@ def parse_for_data(self):
...
-        
+
+        if self.intro_text and self.title:
+            data_elements = {
+                'module': self.title,
+                'function': '',
+                'method_signature': '',
+                'first_paragraph': self.intro_text,
+                'url': self.create_url('')
+                }
```
Comment (gaulrobe) [inline lib/fathead/python/parse.py]: Nit: Bracket should line up under `data_elements`

**Comment by gaulrobe** [comment]
This looks pretty good to me!

File: lib/fathead/python/parse.py (hunk 2 above)
Comment (jdorweiler) [inline lib/fathead/python/parse.py]: Nope, nice catch.

**Comment by jdorweiler** [comment]
Thanks for the review @gaulrobe

**Pull request merged**